### PR TITLE
Fixes following manual tests

### PR DESCRIPTION
**Pull request by ValJed**
## Summary

Fixes did during manual testing.

### CODE REVIEW

File: modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue [hunk 1]
```diff
@@ -19,6 +19,7 @@
         :current="docFields.data"
         :published="published"
         :show-edit="false"
+        :can-delete-draft="moduleOptions.canDeleteDraft"
```
Comment (ValJed) [inline modules/@apostrophecms/doc-type/ui/apos/components/AposDocEditor.vue:22]: passing this info to AposDocEditor context menu (was already working for manager modal items)

File: modules/@apostrophecms/doc-type/ui/apos/logic/AposDocContextMenu.js [hunk 2]
```diff
@@ -254,7 +254,7 @@ export default {
       if (!this.context.lastPublishedAt && !this.canDeleteDraft && !this.context._delete) {
         return false;
       }
-      if (this.context.lastPublishedAt && !this.canEdit) {
+      if (this.context.lastPublishedAt && (!this.context.modified || !this.context._edit)) {
```
Comment (ValJed) [inline modules/@apostrophecms/doc-type/ui/apos/logic/AposDocContextMenu.js:257]: Fixes the logic to show or not the `delete draft` button

File: modules/@apostrophecms/page/index.js [hunk 3]
```diff
@@ -1119,7 +1119,7 @@ database.`);
             // Move outside tree
             throw self.apos.error('forbidden');
           }
-          if ((oldParent._id !== parent._id) && (parent.type !== '@apostrophecms/archive-page') && (!parent._edit)) {
+          if ((oldParent._id !== parent._id) && (parent.type !== '@apostrophecms/archive-page') && (!parent._create)) {
```
Comment (ValJed) [inline modules/@apostrophecms/page/index.js:1122]: Need the create permission to be able to move a page (used to restore it)

File: modules/@apostrophecms/piece-type/index.js [hunk 4]
```diff
@@ -850,7 +850,10 @@ module.exports = {
         return self.findOneForEditing(
           req,
           { _id: piece._id },
-          { attachments: true }
+          {
+            attachments: true,
+            permission: 'create'
+          }
```
Comment (ValJed) [inline modules/@apostrophecms/piece-type/index.js:856]: Checks create permission when inserting a piece.

**Review by boutell — APPROVED** [review]
Approving. The test failure was a fluke out of RAM error looks like, I hit the re-run button.

**Review by haroun — APPROVED** [review]

**Pull request merged**